client/src/pages/Signup.js

```diff
@@ -0,0 +1,155 @@
+import React, { useState, } from 'react';
+import CssBaseline from '@material-ui/core/CssBaseline';
+import Typography from '@material-ui/core/Typography';
+import { makeStyles } from '@material-ui/core/styles';
+import Container from '@material-ui/core/Container';
+import Snackbar from '@material-ui/core/Snackbar';
+import SubmitButton from '../components/SubmitButton'
+import CustomTextField from '../components/CustomTextField'
+import axios from 'axios'
+
+
+const useStyles = makeStyles((theme) => ({
+  paper: {
+    marginTop: theme.spacing(8),
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center',
+  },
+  form: {
+    width: '100%',
+    marginTop: theme.spacing(1),
+  },
+  mainContainer: {
+    border: '1px solid white',
+    borderRadius: '5px',
+    boxShadow: '0 2px 4px 0 gray',
+    backgroundColor: 'white',
+    marginTop: '100px',
+    width: '900px',
+  },
+}));
+
+export default function Signup(props) {
+  const classes = useStyles();
+  const loginSignup = props.location.pathname
+
+  const [email, setEmail] = useState('')
+  const [company, setCompany] = useState('')
+  const [password, setPassword] = useState('')
+  const [open, setOpen] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('')
+
+  const createAccount = async (event) => {
+    try {
+      event.preventDefault()
+      let passedFields = true
+
+      //error handling here for fields
+      if (!validateEmail(email)) {
+        setErrorMessage('Please enter a valid email')
+        passedFields = false
+
+      }
+      if (password.length < 7) {
+        setErrorMessage('Please enter a password longer than 7 characters')
+        passedFields = false
+      }
+      if (email === '' || company === '' || password === '') {
+        setErrorMessage('Please fill in all fields')
+        passedFields = false
+      }
+      if (!passedFields) {
+        setOpen(true)
+        return
+      }
+
+      //if frontend validations pass, make server call here to create user
+      const res = await axios.get('welcome', {
```

Comment:
I think sign up should use "post" instead of "get"?

Reply:
Yes it's post, i am adding some minor things and changed + pushed a commit accidentally during testing. Next commit should fix it!